frontend/src/pages/EditProductPage.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Formik, Form } from 'formik'
+import { useParams, useHistory } from 'react-router-dom'
+import { Button } from 'react-bootstrap'
+import InputField from '../components/InputField'
+import SelectCategory from '../components/listofproducts/SelectCategory'
+import validationEdit from '../communications/validationEdit'
+import { updateProductData, fetchOneProduct } from '../communications/userApi'
+import EditResponseModal from '../components/modals/EditResponseModal'
+
+const updateOldFieldsInItem = (newProductValues, product) => {
+  Object.keys(product).forEach((key) => {
+    return newProductValues[key]
+      ? (product[key] = newProductValues[key])
+      : product[key]
```

Comment:
Ez még mindig szaglik nekem:
1. forEach-ben fölösleges return-olni, az csak map-nél, filter-nél stb. fontos.
2. Az elsőből következően a ternary második ága nem csinál semmit, hiszen a product[key]-t return-öli.
3. A ternary első ága is kiértékelődik valamire (találós kérdés: egy értékadás mire érétkelődik ki?)
4. Az, hogy megtörténik az értékadás csak egy mellékhatás, ha a ternary első ága a nyertes.

Nem ugyenzt csinálja és rövidebb belespreadelni?
`return {...product, ... newProductValues}`

Reply:
Spreadelés nem működik. A return pontosan a második ág miatt volt, mivel az önmagában nem volt értékadás és forEach nem ette meg. Ha már ES6, ez jónak tűnik: newProductValues[key] && (product[key] = newProductValues[key]).
Ha szerkesztéskor nem üres string a mező, azaz truthy, akkor ezzel az új adattal felülírjuk a product egyazon mezejét.